// Copyright 2022 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message ApigeeBetaAttachment {
  string name = 1;
  string environment = 2;
  int64 created_at = 3;
  string envgroup = 4;
}

message ApplyApigeeBetaAttachmentRequest {
  ApigeeBetaAttachment resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteApigeeBetaAttachmentRequest {
  string service_account_file = 1;
  ApigeeBetaAttachment resource = 2;
}

message ListApigeeBetaAttachmentRequest {
  string service_account_file = 1;
  string Envgroup = 2;
}

message ListApigeeBetaAttachmentResponse {
  repeated ApigeeBetaAttachment items = 1;
}

service ApigeeBetaAttachmentService {
  rpc ApplyApigeeBetaAttachment(ApplyApigeeBetaAttachmentRequest) returns (ApigeeBetaAttachment);
  rpc DeleteApigeeBetaAttachment(DeleteApigeeBetaAttachmentRequest) returns (google.protobuf.Empty);
  rpc ListApigeeBetaAttachment(ListApigeeBetaAttachmentRequest) returns (ListApigeeBetaAttachmentResponse);
}
